src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static partial class OperationExtensions
+    {
+        /// <summary>
+        /// Returns the <see cref="ValueUsageInfo"/> for the given operation.
+        /// This extension can be removed once https://github.com/dotnet/roslyn/issues/25057 is implemented.
+        /// </summary>
+        public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
+        {
+            /*
+            |    code         | Read | Write | ReadableRef | WritableRef |
+            | nameof(x)       |      |       |             |             |
```

Comment:
is nameof usage properly handled/tested in your removed-unused-members code?

Reply:
nameof usage is essentially ignored - a member that is only used in nameof can be removed and replaced with a const string assigned to it's own name, and the nameof sites can be replaced directly with that const.

---
In reply to: [212735506](https://github.com/dotnet/roslyn/pull/29511#discussion_r212735506) [](ancestors = 212735506)

Reply:
#unbydesigned :)

If we're going to remove their code, we need to do the thing you mention.  Specifically, we need to replace with such a const string.  We can't remove and then tell teh user: it's ok... just go do that yourself.

Reply:
For now, I am going to avoid flagging this case in the analyzer, and file a separate issue to enhance the code fix.

---
In reply to: [212782505](https://github.com/dotnet/roslyn/pull/29511#discussion_r212782505) [](ancestors = 212782505)

Reply:
This will no longer be flagged now. Filed [https://github.com/dotnet/roslyn/issues/29519](https://github.com/dotnet/roslyn/issues/29519) to track enhancement of code fix.

---
In reply to: [212797081](https://github.com/dotnet/roslyn/pull/29511#discussion_r212797081) [](ancestors = 212797081,212782505)